test/test_changelog.py

```diff
@@ -54,16 +54,37 @@ def test_changelog_content(self):
         parser = ChangelogParser(parent_folder="test/fixtures")
         self.assertIsInstance(parser.CHANGELOG_FILEPATH, Path)
 
-        expected_version_note = (
-            "- This is a version with a prerelease in this changelog\n"
-            "- The changelog module is tested against these lines\n"
-            "- Be careful modifying this file"
-        )
+        # Unreleased doesn't count
+        self.assertEqual(7, len(parser._parse()))
 
-        # checks
+        # This version doesn't exist
         self.assertIsNone(parser.content("0.0.0"))
-        self.assertIsInstance(parser.content("10.1.0-alpha1"), str)
-        self.assertEqual(parser.content("10.1.0-alpha1"), expected_version_note)
+
+        expected_checks = {
+            "10.1.0-beta1": (
+                "- This is the latest documented version in this changelog\n"
+                "- The changelog module is tested against these lines\n"
+                "- Be careful modifying this file"
+            ),
+            "10.1.0-alpha1": (
+                "- This is a version with a prerelease in this changelog\n"
+                "- The changelog module is tested against these lines\n"
+                "- Be careful modifying this file"
+                # "\n" TODO Fixed section is missing
+                # "- trying with a subsection in a version note"
```

Comment:
This is not working with the current regex it seems.
It's in the changelog : https://github.com/opengisch/qgis-plugin-ci/blob/master/test/fixtures/CHANGELOG.md#1010-alpha1---20210110 using a sub section

```
## Added
## Changed
## Removed
```

Reply:
I think we should priorize the compatibility with the conventional Keep a Changelog over specific usage, no?

Reply:
Ok, sorry I did not see that's already merged. Too late! My bad :wink: .